yt_dlp/extractor/bilibili.py

```diff
@@ -268,31 +241,108 @@ def _real_extract(self, url):
+                    bv_id, cid, f['quality'])
+                video_info_ext = self._download_json(playurl, video_id,
+                                                     note='Downloading video info page',
+                                                     headers=headers)
+                if not video_info_ext:
+                    continue
+                video_info_ext = video_info_ext['data']
 
-        title = self._html_search_regex((
-            r'<h1[^>]+title=(["])(?P<content>[^"]+)',
-            r'<h1[^>]+title=([\'])(?P<content>[^\']+)',
-            r'(?s)<h1[^>]*>(?P<content>.+?)</h1>',
-            self._meta_regex('title')
-        ), webpage, 'title', group='content', fatal=False)
+                slices = []
+                for durl in video_info_ext['durl']:
+                    slices.append({
+                        'url': durl['url'],
+                        'filesize': int_or_none(durl['size'])
+                    })
+                ext = f['format']
+                if ext.startswith('flv'):
+                    # flv, flv360, flv720
+                    ext = 'flv'
+
+                filesize = 0
+                for s in slices:
+                    if s['filesize'] is None:
+                        filesize = None
+                    else:
+                        filesize += s['filesize']
+
+                if len(slices) == 0:
+                    continue
+
+                fmt = {
+                    'url': slices[0]['url'],
+                    'ext': ext,
+                    'quality': f['quality'],
+                    'quality_desc': f['display_desc'],
+                    'height': int_or_none(f['display_desc'].rstrip('P')),
+                    'vcodec': f.get('codecs'),
+                    'acodec': None,
+                    'entries': slices,
+                    'filesize': filesize
+                }
+                formats.append(fmt)
+
+            self._sort_formats(formats)
+
+            info.update({
+                'id': video_id,
+                'duration': float_or_none(video_info.get('timelength'), 1000),
+                'http_headers': {
+                    'Referer': url,
+                },
+            })
+
+            # if all formats have same num of slices, rewrite it as multi_video
+            slice_num_set = set(len(f['entries']) for f in formats)
+            if len(slice_num_set) > 1:
+                fallback_fmt = formats[-1]
+                self.report_warning("[Unsupported] Found formats have different num of slices. Fallback to first format %s" % (fallback_fmt['quality_desc']))
+                formats = [fallback_fmt]
+                slice_num = len(fallback_fmt['entries'])
+            else:
+                slice_num = slice_num_set.pop()
+
+            entries = []
+            for idx in range(slice_num):
+                slice_formats = [{**f, **f['entries'][idx]} for f in formats]
```

Comment:
You can't do this. We have very specific metadata keys we allow

Reply:
> You can't do this. We have very specific metadata keys we allow

`            for idx in range(slice_num):
                slice_formats = [{**f} for f in formats]
                for f in slice_formats:
                    f['url'] = f['entries'][idx]['url']
                    f['filesize'] = f['entries'][idx]['filesize']
                    del f['entries']
`

is this ok?
entries only have url and filesize